packages/xo-web/src/xo-app/proxies/index.js

```diff
@@ -25,17 +30,187 @@ import {
+                    id={state.idIpInput}
+                    name='ip'
+                    onChange={effects.onChange}
+                    pattern='[^\s]+'
+                    required={value.networkMode === 'staticIp'}
+                    value={value.ip}
+                  />
+                </Col>
+              </SingleLineRow>
+              <SingleLineRow className='mt-1'>
+                <Col mediumSize={4}>
+                  <Label htmlFor={state.idNetmaskInput}>{_('netmask')}</Label>
+                </Col>
+                <Col mediumSize={8}>
+                  <input
+                    className='form-control'
+                    id={state.idNetmaskInput}
+                    name='netmask'
+                    onChange={effects.onChange}
+                    placeholder={formatMessage(
+                      messages.proxyNetworkNetmaskPlaceHolder
+                    )}
+                    value={value.netmask}
+                  />
+                </Col>
+              </SingleLineRow>
+              <SingleLineRow className='mt-1'>
+                <Col mediumSize={4}>
+                  <Label htmlFor={state.idGatewayInput}>{_('gateway')}</Label>
+                </Col>
+                <Col mediumSize={8}>
+                  <input
+                    className='form-control'
+                    id={state.idGatewayInput}
+                    name='gateway'
+                    onChange={effects.onChange}
+                    pattern='[^\s]+'
+                    required={value.networkMode === 'staticIp'}
+                    value={value.gateway}
+                  />
+                </Col>
+              </SingleLineRow>
+              <SingleLineRow className='mt-1'>
+                <Col mediumSize={4}>
+                  <Label htmlFor={state.idDnsInput}>{_('dns')}</Label>
+                </Col>
+                <Col mediumSize={8}>
+                  <input
+                    className='form-control'
+                    id={state.idDnsInput}
+                    name='dns'
+                    onChange={effects.onChange}
+                    placeholder={formatMessage(
+                      messages.proxyNetworkDnsPlaceHolder
+                    )}
+                    value={value.dns}
+                  />
+                </Col>
+              </SingleLineRow>
+            </div>
+          )}
+        </Container>
+      ),
+    ]),
     header: (
       <span>
         <Icon icon='proxy' /> {_('deployProxy')}
       </span>
     ),
-  }).then(deployProxyAppliance)
+  }).then(({ sr, networkMode, ip, netmask, gateway, dns }) =>
+    deployProxyAppliance(
+      sr,
+      networkMode === 'staticIp'
+        ? {
+            networkConfiguration: {
+              dns: (dns = dns.trim()) === '' ? undefined : dns,
+              gateway,
+              ip,
+              netmask: (netmask = netmask.trim()) === '' ? undefined : netmask,
+            },
+          }
+        : undefined
```

Comment:
Why not handle this directly in the component above?

Reply:
By using the form `handler` property?